applications/FluidDynamicsApplication/python_scripts/adjoint_fluid_solver.py

```diff
@@ -18,13 +18,30 @@ def __init__(self, model, settings):
         self.min_buffer_size = 2
 
     def AddDofs(self):
+        #TODO: Use this when the fixity issue is solved in the GUI
+        # KratosMultiphysics.VariableUtils.AddDofsList(self.GetDofsList(), self.main_model_part)
+
         KratosMultiphysics.VariableUtils().AddDof(KratosCFD.ADJOINT_FLUID_VECTOR_1_X, self.main_model_part)
         KratosMultiphysics.VariableUtils().AddDof(KratosCFD.ADJOINT_FLUID_VECTOR_1_Y, self.main_model_part)
         KratosMultiphysics.VariableUtils().AddDof(KratosCFD.ADJOINT_FLUID_VECTOR_1_Z, self.main_model_part)
         KratosMultiphysics.VariableUtils().AddDof(KratosCFD.ADJOINT_FLUID_SCALAR_1, self.main_model_part)
 
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Adjoint fluid solver DOFs added correctly.")
 
+    def GetDofsList(self):
+        dofs_list = []
+        dofs_list.append("ADJOINT_FLUID_VECTOR_1_X")
```

Comment:
This can be automated a priori, if you do a list of "components = ["X","Y","Z"]" and you iterate over dimension

Reply:
Yest, but I'd keep it as it is for the sake of readability.